controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -789,6 +769,11 @@ func addStorageRelatedAnnotations(obj client.Object, storageClassClaimName, stor
 	annotations[controllers.StorageCephUserTypeAnnotation] = cephUserType
 }
 
+func (r *StorageClassClaimReconciler) own(resource metav1.Object) error {
+	// Ensure storageClassClaim ownership on a resource
+	return ctrl.SetControllerReference(r.storageClassClaim, resource, r.Scheme)
+}
```

Comment:
This is in conflict with the `own` function used in consumer mode which sets an OwerReference and not a controller reference. Either way, you cannot use it as a resource can only have a single controller reference and we share resources between multiple claims (block pools etc.)

Please revert, or provide a different implementation

Reply:
reverted